storage/rocksdb/rdb_datadic.cc

```diff
@@ -780,6 +783,53 @@ int Rdb_key_def::successor(uchar *const packed_tuple, const uint &len) {
   return changed;
 }
 
+bool Rdb_key_def::covers_lookup(TABLE *const table,
+                                const rocksdb::Slice *const unpack_info) const {
+  DBUG_ASSERT(m_index_type == INDEX_TYPE_SECONDARY);
+  DBUG_ASSERT(table->key_info != nullptr);
+  Rdb_string_reader unp_reader = Rdb_string_reader::read_or_empty(unpack_info);
+
+  // Check if unpack_info is corrupted
+  const bool has_unpack_info =
+      unp_reader.remaining_bytes() &&
+      *unp_reader.get_current_ptr() == RDB_UNPACK_DATA_TAG;
+  if (!has_unpack_info || !unp_reader.read(RDB_UNPACK_HEADER_SIZE)) {
+    return false;
+  }
+
+  MY_BITMAP covered_bitmap;
+  my_bitmap_map covered_bits;
+  bitmap_init(&covered_bitmap, &covered_bits, 16, false);
+  if (unp_reader.read_uint16(&covered_bits)) {
+    return false;
+  }
+
+  uint covered_columns = 0;
+  KEY key_info = table->key_info[get_keyno()];
```

Comment:
Updated. In index_init, the mapping is computed for each index and stored in m_lookup_bitmap_vect. In index_init they are cleared. Rdb_key_def::covers_lookup now only needs to read the bitmap from unpack_info and check if the lookup bitmap is a subset.